build.sh

```diff
@@ -76,18 +91,29 @@ if [ -d "${UT_CONTROL_LIB_DIR}" ]; then
     echo "Framework ut-control already exists"
     # ut-control exists so run the makefile from ut
     check_ut_control_revision
-    #make test
+    if [ -d "${CURL_OBJ_DIR}" ]; then
+        echo "Third party library is built for ${TARGET}"
+    else
+        echo "Third party library needs to be built for ${TARGET}"
+        cd ./ut-control
+        #git checkout ${UT_CONTROL_PROJECT_VERSION}
+        git checkout feature/gh38-separate-obj-dir-ut-control
+        ./configure.sh TARGET=${TARGET}
+        cd ../..
+        make TARGET=${TARGET}
+    fi
 else
     if [ "$1" != "no_ut_control" ]; then
         echo "Clone ut_control in ${UT_CONTROL_LIB_DIR}"
         git clone ${TEST_REPO} ut-control
         check_ut_control_revision
         cd ./ut-control
-        git checkout ${UT_CONTROL_PROJECT_VERSION}
-        ./configure.sh
-        #make lib
-        #cd ..
-        #./${0} $@
+        #git checkout ${UT_CONTROL_PROJECT_VERSION}
+        git checkout feature/gh38-separate-obj-dir-ut-control
+        ./configure.sh TARGET=${TARGET}
+        cd ../..
+        make TARGET=${TARGET}
```

Comment:
what happened if the ut-core exits?
i think we not change the user passed argument 
so better to use  ./${0} $@